restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/GeolocationEndpoint.java

```diff
@@ -0,0 +1,752 @@
+            Geolocation geolocation = dao.getGeolocation(new Sid(sid));
+            if (geolocation != null) {
+                secure(account, geolocation.getAccountSid(), SecuredType.SECURED_APP);
+            }
+        } catch (final Exception exception) {
+            return status(UNAUTHORIZED).build();
+        }
+        dao.removeGeolocation(new Sid(sid));
+        return ok().build();
+    }
+
+    public Response putGeolocation(final String accountSid, final MultivaluedMap<String, String> data,
+                                   GeolocationType geolocationType, final MediaType responseType) {
+        Account account;
+        try {
+            account = accountsDao.getAccount(accountSid);
+            secure(account, "RestComm:Create:Geolocation", SecuredType.SECURED_APP);
+        } catch (final Exception exception) {
+            return status(UNAUTHORIZED).build();
+        }
+
+        try {
+            validate(data, geolocationType);
+        } catch (final NullPointerException nullPointerException) {
+            // API compliance check regarding missing mandatory parameters
+            return status(BAD_REQUEST).entity(nullPointerException.getMessage()).build();
+        } catch (final IllegalArgumentException illegalArgumentException) {
+            // API compliance check regarding malformed parameters
+            cause = illegalArgumentException.getMessage();
+            rStatus = responseStatus.Failed.toString();
+        } catch (final UnsupportedOperationException unsupportedOperationException) {
+            // API compliance check regarding parameters not allowed for Immediate type of Geolocation
+            return status(BAD_REQUEST).entity(unsupportedOperationException.getMessage()).build();
+        }
+
+        /*********************************************/
+        /*** Query GMLC for Location Data, stage 1 ***/
+        /*********************************************/
+        try {
+            String targetMSISDN = data.getFirst("DeviceIdentifier");
+            Configuration gmlcConf = configuration.subset("gmlc");
+            String gmlcURI = gmlcConf.getString("gmlc-uri");
+            // Authorization for further stage of the project
+            String gmlcUser = gmlcConf.getString("gmlc-user");
+            String gmlcPassword = gmlcConf.getString("gmlc-password");
+            // Credentials credentials = new UsernamePasswordCredentials(gmlcUser, gmlcPassword);
+            URL url = new URL(gmlcURI + targetMSISDN);
+            HttpClient client = HttpClientBuilder.create().build();
+            HttpGet request = new HttpGet(String.valueOf(url));
+            // Authorization for further stage of the project
+            request.addHeader("User-Agent", gmlcUser);
+            request.addHeader("User-Password", gmlcPassword);
+            HttpResponse response = client.execute(request);
+            BufferedReader br = new BufferedReader(
```

Comment:
@gvagenas, done via commit 63d23bc. Tested successfully manually with generated binaries and all testsuite Junit tests passing too.